### PR TITLE
[TS] LPS-193158

### PR DESCRIPTION
**Pull request by ChrisKian**
https://liferay.atlassian.net/browse/LPS-193158

Resent from https://github.com/liferay-appsec/liferay-portal/pull/1293

I've completely changed the solution to revert [LPS-189151](https://liferay.atlassian.net/browse/LPS-189151), change the "Post" option to "Basic or Post", and bypass strict authentication method checking should Post or Basic be used.  Please let me know if you have any questions, thanks!

@JoyceWang08211, I believe the test will need to be changed.  The original behavior prior to [LPS-189151](https://liferay.atlassian.net/browse/LPS-189151) is fine, however the label changed from "Client Secret Post" to "Client Secret Basic or Post".  Not sure if that matters for your test though.

### CODE REVIEW

File: modules/apps/oauth2-provider/oauth2-provider-web/src/main/resources/META-INF/resources/admin/edit_application_left_column.jsp [hunk 1]
```diff
@@ -29,7 +29,7 @@ OAuth2Application oAuth2Application = oAuth2AdminPortletDisplayContext.getOAuth2
 	</c:if>
 
 	<aui:select helpMessage="client-authentication-method-help" label="Client Authentication Method" name="clientAuthenticationMethod" required="<%= true %>">
-		<aui:option label="Client Secret Post" value="client_secret_post" />
```
Comment (stian-sigvartsen) [inline modules/apps/oauth2-provider/oauth2-provider-web/src/main/resources/META-INF/resources/admin/edit_application_left_column.jsp:33]: @ChrisKian  I think you are reliant on the dropdown having no explicit option selected when customers selected BASIC prior to this fix? That's a potential solution, but maybe risky if we add another new 1st option in  in the future? Did you consider an upgrade process?

**Comment by JoyceWang08211** [comment]
I will not update related OAuth2 tests until the final fix.
https://liferay.atlassian.net/browse/LRQA-82357

cc @gaborlovasdotcom

File: modules/apps/oauth2-provider/oauth2-provider-web/src/main/resources/META-INF/resources/admin/edit_application_left_column.jsp (hunk 1 above)
Comment (ChrisKian) [reply modules/apps/oauth2-provider/oauth2-provider-web/src/main/resources/META-INF/resources/admin/edit_application_left_column.jsp:33]: I've added an upgrade process in the new PR.  Thanks!

**Comment by ChrisKian** [comment]
Resent here: https://github.com/liferay-appsec/liferay-portal/pull/1304

**Pull request closed without merging**